### ISSUE TITLE
addon-info: Add support for Vue

### ISSUE THREAD

**Issue by DrSensor** [storybookjs/storybook#2812]
### Issue details
This issue is feature request for adding support Vue component on `@sotrybook/addon-info`

### Details
In [authoring reusable Vue component](https://vuejs.org/v2/guide/components.html#Authoring-Reusable-Components), types of documentation that need to be added:

- [Props](https://vuejs.org/v2/guide/components.html#Props)
- [Events](https://vuejs.org/v2/guide/components.html#Custom-Events)
- [Slots](https://vuejs.org/v2/guide/components.html#Named-Slots)

One example in this [readme](https://github.com/DrSensor/vue-authoring-template/tree/develop/template#usage)

---
Also, it would be nice if there is feature for choosing which component to be documented, something like

```js
withInfo({ component: MyComponent })(() => /** story */)
```

### Additional
Currently, I'm using storybook to showcase a Vue component in [mvce](https://stackoverflow.com/help/mcve) style like in [here](https://github.com/DrSensor/vue-annotator/tree/master/src/stories). 
Because it's more comfortable writing scenario/use-case in `.vue` SFC format, so I implement it via this way:

1. register each scenario as [story element-tag](https://github.com/DrSensor/vue-annotator/blob/6ce1e3c1ede49f300a6fc0d2ca861f02e06532f2/src/stories/index.js#L61)
2. write [scenarios](https://github.com/DrSensor/vue-annotator/tree/master/src/stories/BasicShape) in `.vue` SFC format
3. register [my reusable component](https://github.com/DrSensor/vue-annotator/blob/6ce1e3c1ede49f300a6fc0d2ca861f02e06532f2/src/stories/BasicShape/Rectangle.vue#L25) on each scenario

In short: 
`src/stories/index.js` <- `src/stories/story/Scenario.vue` <- `src/components/MyComponent.vue`

**Comment by philwolstenholme** [comment]
+1, this would be great

**Comment by Ineigo** [comment]
Not, closing!!!

**Comment by Hypnosphi** [comment]
#1501 should probably be done before adding support for more frameworks

**Comment by danielduan** [comment]
I would imagine info support for vue to be in a new addon instead of adding it to the existing info addon. There are some pretty significant differences between the two frameworks that I think it justifies splitting them.

**Comment by DrSensor** [comment]
Something like `import { withInfo } from '@storybook/addon-info/vue'`?

**Comment by Aaron-Pool** [comment]
Not the author, but just wanted to say that this exists and seems to be being actively developed: https://github.com/pocka/storybook-addon-vue-info

**Comment by zhaozhe0831** [comment]
https://github.com/pocka/storybook-addon-vue-info this's useful, but not enough.

**Comment by MosheAtOwal** [comment]
bump

**Comment by okinp** [comment]
bump

**Comment by igor-dv** [comment]
Linking with #4770.
You can use https://github.com/pocka/storybook-addon-vue-info.

**Comment by Vianney-VICTORIA** [comment]
Hi there !  any plans, release date to add support for Vue ?

**Comment by DrSensor** [comment]
Seems there is no work in progress for this feature. I wonder how the vue storybook will differ from https://vueds.com 🤔

**Comment by Vianney-VICTORIA** [comment]
Thanks for the share @DrSensor, I will take a look !!

**Comment by shilman** [comment]
I have a working prototype for Vue and will be shipping this out in the 5.x cycle. For more information see the RFC and a short demo:

RFC: https://docs.google.com/document/d/1D4Ov4UvrB80Uy1CLGshbFEWQKWEpr8dL4FmOqRDwjwI/edit?usp=sharing
Demo: https://twitter.com/mshilman/status/1113532547764342784

**Comment by elevatebart** [comment]
@shilman, can't wait !!

**Comment by DrSensor** [comment]
@probot waiting for v5.x

**Comment by forforeach** [comment]
Is there anything going on actually? Aren't we in a 5.x yet?

**Comment by shilman** [comment]
@forforeach Vue is supported in the addon-docs technical preview including props, source, docgen info, notes, etc.

https://docs.google.com/document/d/1un6YX7xDKEKl5-MVb-egnOYN8dynb5Hf7mq0hipk8JE/edit?usp=sharing

![image](https://user-images.githubusercontent.com/488689/58715336-79408b80-837b-11e9-9ee7-1e08aa1e4582.png)

**Comment by DrSensor** [comment]
@probot waiting for the sneak peek

**Comment by shilman** [comment]
@DrSensor why wait, it's been available in alpha for over a month?

**Comment by DrSensor** [comment]
Wait what..., yeah 💃 will take a 👀

**Comment by shilman** [comment]
@DrSensor Feel free to join in the conversation on `#docs-mode` in our discord https://discord.gg/UUt2PJb
